feat(curriculum): Build a Zoo Animal Scatterplot (#66379)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-build-a-zoo-animal-scatterplot/69f4597f48697f7e92ddc24b.md

---
id: 69f4597f48697f7e92ddc24b
title: Step 11
challengeType: 0
dashedName: step-11
---

# --description--

In this step, you will learn about the 2nd scale generator `scaleLinear()`.  
*scaleLinear()*  
In the scale, you can map continous domain to a continous range. To understand the above code look carefully to the below diagrams, that map domain value 0 to 100 with range value 0 to 1000.

```js
const f = d3.scaleLinear([0, 100], [0, 1000])
f(10);    // 100 
```

Inside d3.scaleLinear() method, the first array refers the domain and second array refers the range.

It maps 0 -> 0  
        1 -> 10  
        2 -> 20  
        .  
        .  
        50 -> 500  
        .  
        . 
        99 -> 990
        100 -> 1000  

 
 Similarly, you can map inversely, as example domain value 0 to 100 with range value 1000 to 0. 

```js
const f = d3.scaleLinear([0, 100], [1000, 0])
f(10)     //900
```

It maps 0 -> 1000  
        1 -> 990  
        2 -> 980  
        .  
        .  
        50 -> 500  
        .  
        .  
        99 -> 10
        100 -> 0  

> Now, with this understanding create a linear scale `x` that maps domain 0 to 100 with range 0 to `width` of content.

Reviewer: ````suggestion
Unlike `scaleOrdinal()`, which maps categories to values, `scaleLinear()` is used to map continuous numeric values. 

Here is an example of converting values from `0` to `100` into pixel positions between `0` and `width` so elements can be positioned correctly across the SVG.

```js
const f = d3.scaleLinear([0, 100], [0, 1000]);

f(10); // 100
```

In `d3.scaleLinear()`, the first array defines the domain and the second array defines the range.

You can also reverse the mapping:

```js
const f = d3.scaleLinear([0, 100], [1000, 0]);

f(10); // 900
```

Create a linear scale named `x` that maps the domain `[0, 100]` to the range `[0, width]`.

````

# --hints--

> You should create a Linear scale `x`.

Reviewer: ```suggestion
You should created a variable called `x` and assign it `d3.scaleLinear()`
```

```js
assert.match(
  code,

>   /const\s+x\s*=\s*d3\.scaleLinear\([\s\S]*\)/

Reviewer: `let` should be accepted here in these tests

);
```

> Your `x` scale should have domain value 0 to 100 and range value 0 to width.

Reviewer: ```suggestion
Your `x` scale should have domain values from `0` to `100` and range values from `0` to `width`.
```
